### PR TITLE
[1880] Start companies

### PR DESCRIPTION
**Pull request by crericha**

### CODE REVIEW

File: lib/engine/game/g_1880/game.rb [hunk 1]
```diff
@@ -167,6 +177,73 @@ class Game < Game::Base
...
+          @corporation_building_permits = {}
+          @active_building_permit = 'A'
+          @building_permit_choices_by_president_percent = {
+            20 => %w[ABC BCD],
+            30 => %w[AB BC CD],
+            40 => %w[A B C D],
+          }
+        end
+
+        def setup_unsaleable_shares
```
Comment (bentziaxl) [inline lib/engine/game/g_1880/game.rb:243]: did you merge this with the setup method in my PR?

https://github.com/tobymao/18xx/blob/f3ee29ccd2bdaf82f7881e1ddad146012ecc095a/lib/engine/game/g_1880/game.rb#L259

**Comment by bentziaxl** [comment]
Can you add the permit info to the corp view box, similar to 1862?
image

using  def status_array(corp) ?

**Comment by bentziaxl** [comment]
did you implement that requirement that for floating/par percentages, only shares owned by the president counts?
(you marked it as complete in the checklist, not sure where it's defined in the code)

File: lib/engine/game/g_1880/game.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/game/g_1880/game.rb:243]: No, I'll merge our PRs once your PR is merged into master

**Comment by crericha** [comment]
> Can you add the permit info to the corp view box, similar to 1862? image
> 
> using def status_array(corp) ?

Okay, I will change from using an ability to the status_array

**Comment by crericha** [comment]
> did you implement that requirement that for floating/par percentages, only shares owned by the president count? (you marked it as complete in the checklist, not sure where it's defined in the code)

No, I split that item into two items. I'll add the floating logic when I make the other changes from the review.

**Pull request merged**